front/src/applications/editor/tools/rangeEdition/components/RangeEditionLeftPanel.tsx

```diff
@@ -211,122 +219,129 @@ const RangeEditionLeftPanel = () => {
+            />
+          )}
+        </div>
+        <hr />
+        {isSpeedRestriction && (
+          <>
+            {t('Editor.tools.speed-edition.select-switches-to-get-route')}
+            {Object.keys(selectedSwitches).length > 0 && (
+              <>
+                <SwitchList
+                  selectedSwitches={selectedSwitches}
+                  unselectSwitch={unselectSwitch}
+                  setSwitchSelection={setState}
+                  availableSwitchesPositions={availableSwitchesPositions}
                 />
-              </div>
+                {switchesRouteCandidates.length === 0 && (
+                  <p className="text-muted">{t('Editor.tools.routes-edition.routes_zero')}</p>
+                )}
+              </>
             )}
-            {!speedRestriction && isPSL && (
-              <EditPSLSection
-                entity={entity as SpeedSectionPslEntity}
-                setState={
-                  setState as (
-                    stateOrReducer: PartialOrReducer<RangeEditionState<SpeedSectionEntity>>
-                  ) => void
-                }
-                trackSectionsCache={trackSectionsCache}
-              />
-            )}
-          </div>
-          <hr />
-        </>
-      )}
-      {speedRestriction && (
-        <>
-          {t('Editor.tools.speed-edition.select-switches-to-get-route')}
-          {Object.keys(selectedSwitches).length > 0 && (
-            <>
-              <SwitchList
-                selectedSwitches={selectedSwitches}
-                unselectSwitch={unselectSwitch}
-                setSwitchSelection={setState}
-                availableSwitchesPositions={availableSwitchesPositions}
-              />
-              {switchesRouteCandidates.length === 0 && (
-                <p className="text-muted">{t('Editor.tools.routes-edition.routes_zero')}</p>
-              )}
-            </>
-          )}
-          <hr />
-        </>
-      )}
-      {switchesRouteCandidates.length > 0 && (
-        <>
+            <hr />
+          </>
+        )}
+        {switchesRouteCandidates.length > 0 && (
           <RouteList
             switchesRouteCandidates={switchesRouteCandidates}
             onRouteSelect={handleRouteClicked(true)}
             selectedRoutes={entity.properties.on_routes || []}
             onRouteHighlight={handleRouteClicked(false)}
             highlightedRoutes={highlightedRoutes}
           />
-          <TrackRangesList />
-        </>
-      )}
+        )}
+        <TrackRangesList speedRestrictionTool={isSpeedRestriction} />
+        {!isNew && <EntityError className="mt-1" entity={entity} />}
+      </div>
+    );
+  }
 
+  // Entity is electrification
+  return (
+    <div>
+      <legend className="mb-4">{t(`Editor.obj-types.Electrification`)}</legend>
+      {voltages && <ElectrificationMetadataForm voltages={voltages} />}
+      <hr />
+      <TrackRangesList speedRestrictionTool={isSpeedRestriction} />
```

Comment:
Make speedRestrictionTool nullable and remove the props here, since a electrification will never be a speedRestriction
```suggestion
      <TrackRangesList />
```

Reply:
done